### PR TITLE
Add space if needed when inserting shortcut

### PR DESCRIPTION
**Pull request by danlee1025**
_Pull requests into Flux require the following. Submitter should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:. Reviewers should complete every item on the bulleted list when reviewing._

Adds space when inserting a shortcut if there is no space before it.

## Submitter:

**Running the Application**

- [x] Manually tested in Chrome (primary browser for testing)
- [x] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [x] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [x] Branch implements task as expected
-  Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [x] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [x] This pull request describes why these changes were made
-  Recognizes any potential shortcomings/bugs in the description 
-  Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
-  Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
-  Document any new APIs/extension points
-  Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [x] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [x] Code is commented

**Tests**

- [x] Existing tests passed
-  Added backend tests for new functionality added

## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

### CODE REVIEW

File: src/notes/SingleHashtagKeywordStructuredFieldPlugin.jsx [hunk 1]
```diff
@@ -12,6 +12,9 @@ function SingleHashtagKeywordStructuredFieldPlugin(opts) {
     const insertStructuredFieldTransform = opts.insertStructuredFieldTransform;
     const structuredFieldMapManager = opts.structuredFieldMapManager;
     function onBeforeInput(e, data, editorState) {
+        // short circuit if character typed is not a space
+        if (e.data !== ' ') return;
```
Comment (Dtphelan1) [inline src/notes/SingleHashtagKeywordStructuredFieldPlugin.jsx]: (Not that we have even tested voice in a _minute_, butttt) This might brick voice (again, in case we haven't already bricked it with our many other changes)

**Comment by Dtphelan1** [comment]
Less a task necessarily and more a question: it looks like we have to do that spacing in two different locations - the structuredfieldplugin and in the onBeforeInput. Any reason why just one of those didn't work?

File: src/notes/SingleHashtagKeywordStructuredFieldPlugin.jsx (hunk 1 above)
Comment (danlee1025) [reply src/notes/SingleHashtagKeywordStructuredFieldPlugin.jsx]: Hmm ok will remove it... was unsure if I wanted to include the short circuit in this PR.

**Comment by danlee1025** [comment]
This was the part that I struggled with a little bit.  The check in `StructuredFieldPlugin.jsx` solves the spacing issue in 90% of cases but when we are auto-detecting keywords it fails to recognize that there is no space when we go into the `insertStructuredField` function in `StructuredFieldPlugin.jsx`.  Matt and I discussed and thought this was an OK solution.  Let me know if you think this is a little too hacky and I can debug further.

**Comment by gregquinn2001** [comment]
reworking those 2 cases is not part of this task Dan so I think you're fine. Also, dictation is broken no doubt. We don't test it so don't worry about it.  Maybe leave the code that might work for it commented out so we have it still quickly but add a comment clearly indicating it's for Dragon Naturally Speaking dictation support. We should probably consider a new task to try to support the latest version of the dragon software. We may be getting that in the lab through our potential work with Nuance.

**Comment by Dtphelan1** [comment]
That sounds good to me Greg! 

@danlee1025  if you needed the short circuit don't take it out on my account. Like Greg pointed out this falls outside your scope and doesn't even definitively break something.

**Comment by Dtphelan1** [comment]
[DP1] - Doesn't look like the space gets added when there's nothing but plaintext in the editor. To replicate: 
1. Create new note.
2. Type "anything".
3. Insert @patient with no space.

**Comment by danlee1025** [comment]
DP1 should be fixed now.

**Comment by Dtphelan1** [comment]
The smallest of grievances... [DP2] - When I insert a shortcut on a newline (though not the first line), it inserts a space even though my shortcut should be the first word in the paragraph.

1. New note
2. Type "For Reference"
3. Hit enter for a newline.
4. Insert @patient.

You can see that the F in "For Reference" and the first character in @patient are not aligned.

**Comment by danlee1025** [comment]
Good catch, I'll look into this.

**Comment by danlee1025** [comment]
DP2 should be fixed now.
